cdp_scrapers/instances/portland-static.json

```diff
@@ -8,69 +45,63 @@
-                "external_source_id": null,
-                "roles": [
-                    {
-                        "title":  "Council President",
-                        "start_datetime": 1483257600,
-                        "end_datetime": 1735632000,
-                        "body": {"name": "City Council"}
-                    },
-                    {
-                        "title":  "Member",
-                        "start_datetime": 1483257600,
-                        "end_datetime": 1735632000,
-                        "body": {"name": "City Budget"}
-                    },
-                    {
-                        "title":  "Member",
-                        "start_datetime": 1483257600,
-                        "end_datetime": 1735632000,
-                        "body": {"name": "Management and Finance"}
-                    },
-                    {
-                        "title":  "Member",
-                        "start_datetime": 1483257600,
-                        "end_datetime": 1735632000,
-                        "body": {"name": "Government Relations"}
-                    },
-                    {
-                        "title":  "Member",
-                        "start_datetime": 1483257600,
-                        "end_datetime": 1735632000,
-                        "body": {"name": "City Attorney"}
-                    },
-                    {
-                        "title":  "Member",
-                        "start_datetime": 1483257600,
-                        "end_datetime": 1735632000,
-                        "body": {"name": "Police"}
-                    },
-                    {
-                        "title":  "Member",
-                        "start_datetime": 1483257600,
-                        "end_datetime": 1735632000,
-                        "body": {"name": "Prosper Portland"}
-                    },
-                    {
-                        "title":  "Member",
-                        "start_datetime": 1483257600,
-                        "end_datetime": 1735632000,
-                        "body": {"name": "Emergency Management"}
-                    },
-                    {
-                        "title":  "Member",
-                        "start_datetime": 1483257600,
-                        "end_datetime": 1735632000,
-                        "body": {"name": "Equity and Human Rights"}
-                    }
-                ]
-            },
+            "seat": "Mayor",
+            "roles": [
+                {
+                    "title":  "Council President",
+                    "start_datetime": 1483257600,
+                    "end_datetime": 1735632000,
+                    "body": "City Council"
+                },
+                {
+                    "title":  "Member",
+                    "start_datetime": 1483257600,
+                    "end_datetime": 1735632000,
+                    "body": {"name": "City Budget"}
```

Comment:
Sorry To, I do not understand why only one of them would end up in the DB. :sweat_smile:  Once these entries are ingested to `Roles`, they are just items `i` and `j` in that `Person.seat.roles`.

Reply:
The ingestion role, both the one in the static file and the one retrieved from legistar) will have the same hash (at least according to the DB data model `Role` primary keys -- hash are created from the primary keys). So, only one of them will be in the DB.

Reply:
Thanks, now I understand your original point. In other words, `Person.seat.roles` is a list not a set. My guess would be that the last one in the list wins as the one that gets kept as the `database.models.Role` in the DB among the duplicates (same "title", "person_ref", "body_ref", "seat_ref" if I'm understanding things correctly), whether it "came" from the static data file or a Legistar OfficeRecord.

Reply:
But, in `sanitize_roles()`, I discard a scraped `

Reply:
However, we will drop a dynamically scraped `Role` on the floor if its `body` is in `primary_bodies` from the static data file:
https://github.com/dphoria/cdp-scrapers/blob/0b6731af0e29be007562af7c0192c155bcf50c0f/cdp_scrapers/scraper_utils.py#L289-L297

```Python
def sanitize_roles():
    """
    ...
    Notes
    -----
    Remove roles[#] if roles[#].body in static_data.primary_bodies.
    Use known_static_data.persons[#].seat.roles instead.
    """
    ...
```

So that should lower redundancy a little bit.

Reply:
Agree. Just pointing out that the portland static file has non-primary roles. This means both these roles and the roles retrieved from legistar (also non-primary) will be ingested. Hopefully, no duplicates will make it into DB.